dts/bindings/clock/renesas,ra-cgc-pll-out.yaml

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2024 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Renesas RA Clock Generation Circuit PLL Clock out line
+
+compatible: "renesas,ra-cgc-pll-out"
+
+include: [clock-controller.yaml, base.yaml]
+
+properties:
+  div:
+    required: true
+    type: int
+  freq:
```

Comment:
The frequency originally input to the PLL is multiplied and divided by PLLMUL and PLLDIV, and the divided frequency is then further divided by P, Q, and R to determine the frequency.
Is the frequency set here reflecting PLLMUL and PLLDIV?
I would like to find this calculately if possible, but if it becomes too complicated I think this is fine.

Reply:
@soburi , The fix is not relate to the calculation of PLL P Q R but to separate them as device node so that other node can choose them as clock source, the previous patch only allow the node to choose <&pll> as clock source, now we can set <&pllp>, <&pllq>, <&pllr>. Frequency calculation is as your understanding.
Your request here is to create a macro for the frequency calculation instead of set them on device tree, is my understanding correct ?

Reply:
@duynguyenxa 

I understood why the change is necessary.
This change separates PLLP, Q, and R into independent nodes, so I think it becomes necessary to correctly express their relationship with the PLL.

What I'm paying attention to is whether the device tree give an appropriate expression for the hardware configuration.

From the register configuration, it appears that PLLP, Q, and R are dividers that use the PLL as their source, so I don't think they have frequency attributes here, these has only 'ratio'.
Instead, I think we need to calculate the relationship to the PLL.
This is why we need a frequency calculation macro.

I think that a "fixed-factor-clock" can be used for this node.

Reply:
@soburi , I got it, I think for each clock node here we will need a clock driver to provide enough information for them,
current clock cannot get pll frequency so now, to keep it simple I would like to keep this implementation,
Next patch for clock driver we will update driver for each clock node and change PLL Q P R into fixed-factor-clock.

Reply:
I think there is no problem with proceeding as is, but if possible, it may be preferable to incorporate this change into 4.0.0 as well, since this would not result in any changes to the external interface.